### PR TITLE
Fix assert in token refresh

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
Fixes https://github.com/firebase/firebase-android-sdk/issues/3074

### CODE REVIEW

File: firebase-database/src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java [hunk 1]
```diff
@@ -761,17 +761,11 @@ private void tryScheduleReconnect() {
                 .addOnSuccessListener(
                     executorService,
                     aVoid -> {
-                      // Someone could have interrupted us while fetching the token,
-                      // marking the connection as Disconnected
-                      if (connectionState == ConnectionState.GettingToken) {
-                        if (thisGetTokenAttempt == currentGetTokenAttempt) {
+                      if (thisGetTokenAttempt == currentGetTokenAttempt) {
```
Comment (schmidt-sebastian) [inline firebase-database/src/main/java/com/google/firebase/database/connection/PersistentConnectionImpl.java:764]: This aligns the code back with the pre-AppCheck flow: https://github.com/firebase/firebase-android-sdk/commit/6da8617afe903126afac46ef71b1d857dcfe7ef9#diff-ff27763e8a3b1c249d9220a66053d6ca0dcdd3314480a2eb756a07d7fdf046c9L720

**Review by jsdt — APPROVED** [review]

**Pull request merged**